test/e2e/dra/dra.go

```diff
@@ -1820,6 +1820,212 @@ var _ = framework.SIGDescribe("node")(framework.WithLabel("DRA"), func() {
+		// The third pod requests extended resource by device plugin.
+		f.It("must run pods with extended resource on dra nodes and device plugin nodes", f.WithSerial(), func(ctx context.Context) {
+			pod := b.Pod()
+			res := v1.ResourceList{}
+			res[v1.ResourceName(drautils.ExtendedResourceName(-1))] = resource.MustParse("1")
+			pod.Spec.Containers[0].Resources.Requests = res
+			pod.Spec.Containers[0].Resources.Limits = res
+
+			b.Create(ctx, pod)
+			err := e2epod.WaitForPodRunningInNamespace(ctx, f.ClientSet, pod)
+			framework.ExpectNoError(err, "start pod")
+			containerEnv := []string{
+				"container_0_request_0", "true",
+			}
+			drautils.TestContainerEnv(ctx, f, pod, pod.Spec.Containers[0].Name, false, containerEnv...)
+
+			pod1 := b.Pod()
+			pod1.Spec.Containers[0].Resources.Requests = res
+			pod1.Spec.Containers[0].Resources.Limits = res
+
+			b.Create(ctx, pod1)
+			err = e2epod.WaitForPodRunningInNamespace(ctx, f.ClientSet, pod1)
+			framework.ExpectNoError(err, "start pod1")
+			drautils.TestContainerEnv(ctx, f, pod1, pod.Spec.Containers[0].Name, false, containerEnv...)
+
+			nodes, err := f.ClientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+			framework.ExpectNoError(err, "getting cluster nodes")
+			var availableNodes []string
+			for _, node := range nodes.Items {
+				if _, ok := driver.Nodes[node.Name]; ok {
+					continue
+				}
+				availableNodes = append(availableNodes, node.Name)
+			}
+			extendedResourceName := deployDevicePlugin(ctx, f, availableNodes)
+			// drautils.ExtendedResourceName(-1) must be the same as the returned extendedResourceName
+			// drautils.ExtendedResourceName(-1) is used for DRA drivers
+			// extendedResourceName is used for device plugin.
+			gomega.Expect(string(extendedResourceName)).To(gomega.Equal(drautils.ExtendedResourceName(-1)))
+			ginkgo.DeferCleanup(func(ctx context.Context) {
+				for _, nodeName := range availableNodes {
+					retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+						gotNode, err := f.ClientSet.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
+						if err != nil {
+							return fmt.Errorf("get node %s to update status capacity: %w", nodeName, err)
+						}
+
+						delete(gotNode.Status.Capacity, v1.ResourceName(extendedResourceName))
+						delete(gotNode.Status.Allocatable, v1.ResourceName(extendedResourceName))
+						if _, err := f.ClientSet.CoreV1().Nodes().UpdateStatus(ctx, gotNode, metav1.UpdateOptions{}); err != nil {
+							return fmt.Errorf("update node %s status capacity:  %w", gotNode.Name, err)
+						}
+						return nil
+					})
+					framework.ExpectNoError(retryErr, "update node status capacity")
+				}
+			})
+
+			pod2 := b.Pod()
+			pod2.Spec.Containers[0].Resources.Requests = res
+			pod2.Spec.Containers[0].Resources.Limits = res
+
+			b.Create(ctx, pod2)
+			err = e2epod.WaitForPodRunningInNamespace(ctx, f.ClientSet, pod2)
+			framework.ExpectNoError(err, "start pod2")
```

Comment:
Can you add some verification here that the pod is not using DRA, which is the expected outcome? For example, check the pod status?

Reply:
A check that the special pod status is empty would be cheap and validate further that really happened what was expected.